src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1079,7 +1079,25 @@ private static int IndexOfInitializerContainingPosition(ImmutableArray<FieldOrPr
 
         public override IEnumerable<string> MemberNames
         {
-            get { return this.declaration.MemberNames; }
+            get
+            {
+                return IsTupleType ? tupleMemberNames(GetMembers()) : this.declaration.MemberNames;
```

Comment:
>tupleMemberNames(GetMembers()) [](start = 37, length = 30)

`GetMembers().Distinct()`

Reply:
I think the member names are ordered.

The old implementation used `GetMembers()`:
``` csharp
  // from TupleTypeSymbol.cs
  public override IEnumerable<string> MemberNames
        {
            get
            {
                var set = PooledHashSet<string>.GetInstance();
                foreach (var member in GetMembers())
                {
                    var name = member.Name;
                    if (set.Add(name))
                    {
                        yield return name;
                    }
                }

                set.Free();
            }
        }
```

---
In reply to: [355576198](https://github.com/dotnet/roslyn/pull/39370#discussion_r355576198) [](ancestors = 355576198)